app/src/main/java/com/whereuat/whereu/MainActivity.java

```diff
@@ -96,6 +109,23 @@ public void onReceive(Context context, Intent intent) {
             Toast.makeText(context, text, Toast.LENGTH_SHORT).show();
         }
     }
+
+    public void addContact(View view){
+
+        // Check the SDK version and whether the permission is already granted or not.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && checkSelfPermission(Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.READ_CONTACTS}, PERMISSIONS_REQUEST_READ_CONTACTS);
+        } else {
+            Intent intent= new Intent(Intent.ACTION_PICK,  ContactsContract.Contacts.CONTENT_URI);
+            startActivityForResult(intent, 1);
+        }
+    }
+
+    @Override
+    public void onActivityResult(int reqCode, int resultCode, Intent data) {
+        super.onActivityResult(reqCode, resultCode, data);
+        ContactRetriever con = new ContactRetriever(reqCode,resultCode, data, this.getApplicationContext());
```

Comment:
Can you add a TODO comment above this line to say what it's used for or will be used for.